ui/apps/platform/src/services/ComplianceService.ts

```diff
@@ -10,19 +12,38 @@ export type ComplianceStandardMetadata = {
     numImplementedChecks: number; // int32
     scopes: ComplianceStandardScope[];
     dynamic: boolean;
-    hideScanResults?: boolean; // TODO optional until backend implements new property
+    hideScanResults: boolean;
 };
 
 export function fetchComplianceStandards(): Promise<ComplianceStandardMetadata[]> {
     return axios
-        .get<{ standards: ComplianceStandardMetadata[] }>('/v1/compliance/standards')
-        .then((response) => {
-            return response.data?.standards ?? [];
-        });
+        .get<{ standards: ComplianceStandardMetadata[] }>(standardsUrl)
+        .then((response) => response.data?.standards ?? []);
+}
+
+/*
+ * Temporary frontend work-around for 4.1 version,
+ * because standards are not always sorted in ascending order by name.
+ */
+
+function compareStandardsByName(
+    standardPrev: ComplianceStandardMetadata,
+    standardNext: ComplianceStandardMetadata
+) {
+    const { name: namePrev } = standardPrev;
+    const { name: nameNext } = standardNext;
+
+    /* eslint-disable no-nested-ternary */
+    return namePrev < nameNext ? -1 : namePrev > nameNext ? 1 : 0;
+    /* eslint-enable no-nested-ternary */
+}
+
+export function fetchComplianceStandardsSortedByName(): Promise<ComplianceStandardMetadata[]> {
+    return fetchComplianceStandards().then((standards) => standards.sort(compareStandardsByName));
```

Comment:
Do we need a fallback here, like we have in the regular fetch function above?

Also here, if standards comes back as falsey, or at least not an array, we would need optional chaining, too.

```suggestion
export function fetchComplianceStandardsSortedByName(): Promise<ComplianceStandardMetadata[]> {
    return fetchComplianceStandards().then((standards) => standards?.sort(compareStandardsByName)) ?? [];
```

Reply:
Good question. Before next pull request, I will verify that line 21 in called function is sufficient:

```js
.then((response) => response.data?.standards ?? [])
```